src/Nervos/InputSelector.cpp

```diff
@@ -0,0 +1,185 @@
+    maxWithXInputs.push_back(0);
+    int64_t maxSum = 0;
+    for (auto i = 0; i < n; ++i) {
+        maxSum += sorted[n - 1 - i].capacity;
+        maxWithXInputs.push_back(maxSum);
+    }
+
+    // difference from 2x targetValue
+    auto distFrom2x = [doubleTargetValue](int64_t val) -> int64_t {
+        if (val > doubleTargetValue) {
+            return val - doubleTargetValue;
+        }
+        return doubleTargetValue - val;
+    };
+
+    const int64_t dustThreshold = sizeOfSingleInputAndWitness * byteFee;
+
+    // 1. Find a combination of the fewest inputs that is
+    //    (1) bigger than what we need
+    //    (2) closer to 2x the amount,
+    //    (3) and does not produce dust change.
+    for (size_t numInputs = 1; numInputs <= n; ++numInputs) {
+        const auto fee = (sizeWithoutInputs + numInputs * sizeOfSingleInputAndWitness) * byteFee;
+        const auto targetWithFeeAndDust = targetValue + fee + dustThreshold;
+        if (maxWithXInputs[numInputs] < targetWithFeeAndDust) {
+            // no way to satisfy with only numInputs inputs, skip
+            continue;
+        }
+        auto slices = slice(sorted, static_cast<size_t>(numInputs));
+
+        slices.erase(std::remove_if(slices.begin(), slices.end(),
+                                    [targetWithFeeAndDust](const Cells& slice) {
+                                        return sum(slice) < targetWithFeeAndDust;
+                                    }),
+                     slices.end());
+        if (!slices.empty()) {
+            std::sort(slices.begin(), slices.end(),
+                      [distFrom2x](const Cells& lhs, const Cells& rhs) {
+                          return distFrom2x(sum(lhs)) < distFrom2x(sum(rhs));
+                      });
+            return filterOutDust(slices.front(), sizeOfSingleInputAndWitness, byteFee);
+        }
+    }
+
+    // 2. If not, find a valid combination of outputs even if they produce dust change.
+    for (size_t numInputs = 1; numInputs <= n; ++numInputs) {
+        const auto fee = (sizeWithoutInputs + numInputs * sizeOfSingleInputAndWitness) * byteFee;
+        const auto targetWithFee = targetValue + fee;
+        if (maxWithXInputs[numInputs] < targetWithFee) {
+            // no way to satisfy with only numInputs inputs, skip
+            continue;
+        }
+        auto slices = slice(sorted, static_cast<size_t>(numInputs));
+        slices.erase(std::remove_if(slices.begin(), slices.end(),
+                                    [targetWithFee](const Cells& slice) {
+                                        return sum(slice) < targetWithFee;
+                                    }),
+                     slices.end());
+        if (!slices.empty()) {
+            return filterOutDust(slices.front(), sizeOfSingleInputAndWitness, byteFee);
+        }
+    }
+
+    return {};
+}
+
+Cells InputSelector::selectSimple(int64_t targetValue, int64_t sizeWithoutInputs,
```

Comment:
maybe selectSimple is enough?

Reply:
Yes, implemented logic as per Neuron wallet (Nervos's official wallet).